db/skipped_sequence_test.go

```diff
@@ -0,0 +1,411 @@
+			require.Error(t, err)
+		})
+	}
+}
+
+func BenchmarkRemoveSeqFromSkippedList(b *testing.B) {
+	benchmarks := []struct {
+		name         string
+		rangeEntries bool
+		largeSlice   bool
+	}{
+		{name: "single_entries_large_slice", rangeEntries: false, largeSlice: true},
+		{name: "single_entries_small_slice", rangeEntries: false, largeSlice: false},
+	}
+	for _, bm := range benchmarks {
+		b.Run(bm.name, func(b *testing.B) {
+			skippedSlice := NewSkippedSequenceSlice(DefaultClipCapacityHeadroom)
+			if bm.largeSlice {
+				for i := 0; i < 10000; i++ {
+					skippedSlice.PushSkippedSequenceEntry(NewSingleSkippedSequenceEntry(uint64(i * 2)))
+				}
+			} else {
+				for i := 0; i < 100; i++ {
+					skippedSlice.PushSkippedSequenceEntry(NewSingleSkippedSequenceEntry(uint64(i * 2)))
+				}
+			}
+			for i := 0; i < b.N; i++ {
+				_ = skippedSlice.removeSeq(uint64(i * 2))
+			}
+		})
+	}
+}
+
+// TestInsertItemInSlice:
+//   - Create skipped sequence slice
+//   - Insert a new value in the slice at index 2 to maintain order
+//   - Assert the resulting slice is correct
+//
+// Will add more test cases to this as development continues (for sequence ranges + mixed single and ranges) pending CBG-3853
+func TestInsertItemInSlice(t *testing.T) {
+	testCases := []struct {
+		name         string
+		inputList    []uint64
+		expectedList []uint64
+		insert       uint64
+		index        int
+	}{
+		{
+			name:         "single_items",
+			inputList:    []uint64{2, 6, 100, 200, 500},
+			expectedList: []uint64{2, 6, 70, 100, 200, 500},
+			insert:       70,
+			index:        2,
+		},
+	}
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			skippedSlice := NewSkippedSequenceSlice(DefaultClipCapacityHeadroom)
+
+			for _, input := range testCase.inputList {
+				skippedSlice.PushSkippedSequenceEntry(NewSingleSkippedSequenceEntry(input))
+			}
+
+			// attempt to insert at test case index to keep order
+			skippedSlice.insert(testCase.index, NewSingleSkippedSequenceEntry(testCase.insert))
+
+			for i := 0; i < len(skippedSlice.list); i++ {
+				assert.Equal(t, testCase.expectedList[i], skippedSlice.list[i].getLastSeq())
+			}
+		})
+	}
+}
+
+func BenchmarkInsertSkippedItem(b *testing.B) {
+	benchmarks := []struct {
+		name         string
+		rangeEntries bool
+		largeSlice   bool
+	}{
+		{name: "single_entries_large_slice", rangeEntries: false, largeSlice: true},
+		{name: "single_entries_small_slice", rangeEntries: false, largeSlice: false},
+	}
+	for _, bm := range benchmarks {
+		b.Run(bm.name, func(b *testing.B) {
+			skippedSlice := NewSkippedSequenceSlice(DefaultClipCapacityHeadroom)
+			if bm.largeSlice {
+				for i := 0; i < 10000; i++ {
+					skippedSlice.PushSkippedSequenceEntry(NewSingleSkippedSequenceEntry(uint64(i * 2)))
+				}
+			} else {
+				for i := 0; i < 100; i++ {
+					skippedSlice.PushSkippedSequenceEntry(NewSingleSkippedSequenceEntry(uint64(i * 2)))
+				}
+			}
+			sequenceNum := 40000
```

Comment:
Lines 246-256 look like setup for the benchmark that should be outside of b.Run, since what we're trying to benchmark here is the insert operation.